### PR TITLE
PIM-5763: Add Boolean presenter for fields

### PR DESCRIPTION
**Pull request by pierallard**
**Description (for Contributor and Core Developer)**

This PR add a presenter Boolean, to present data like "enabled". This is done by using a new Presenter type (field), because enabled is not a standard attribute.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | y
| Added Behats                      | y
| Changelog updated                 | n
| Review and 2 GTM                  |
| Micro Demo to the PO (Story only) |
| Migration script                  | n
| Tech Doc                          | n

### CODE REVIEW

**Comment by phaseinducer** [comment]
GTM, perfect :ok_hand:

File: src/Pim/Component/Catalog/Localization/Presenter/PresenterRegistryInterface.php [hunk 1]
```diff
@@ -33,6 +33,15 @@ public function register(PresenterInterface $presenter, $type);
     public function getPresenterByAttributeCode($code);
 
     /**
+     * Get the first presenter supporting a field code
+     *
+     * @param $code
```
Comment (solivier) [inline src/Pim/Component/Catalog/Localization/Presenter/PresenterRegistryInterface.php]: missing type

**Comment by solivier** [comment]
gtm

File: src/Pim/Component/Catalog/Localization/Presenter/PresenterRegistry.php [hunk 2]
```diff
@@ -46,6 +49,19 @@ public function register(PresenterInterface $presenter, $type)
...
      * {@inheritdoc}
      */
+    public function getPresenterByFieldCode($code)
+    {
+        $result = $this->getPresenter($code, self::TYPE_PRODUCT_FIELD);
+        if (null !== $result) {
+            return $result;
+        }
+
+        return $this->getPresenterByAttributeCode($code);
```
Comment (phaseinducer) [inline src/Pim/Component/Catalog/Localization/Presenter/PresenterRegistry.php]: Not a good idea imho, the two types must be separated and you should know which `getPresenter` method to call.

**Comment by pierallard** [comment]
Merged in https://github.com/akeneo/pim-community-dev/pull/4410

**Pull request closed without merging**